#1 General features of 3D genome organization 

## 1.1 2D *cis* elements in the genome 
### Biochemically active regulatory elements (bound by sequence-specific regulatory TFs) :
- **Promoter**: The promoter is a region around the TSS (+1) of a gene, which contains several DNA elements that facilitate the binding of regulatory proteins. It provide a secure initial binding site for RNA polymerase and for proteins called transcription factors that recruit RNA polymerase to make transcription take place [[1]](https://en.wikipedia.org/wiki/Promoter_\(genetics\)).
- **Enhancer**: Enhancers are CRE(cis-regulatory elements which means they are non-coding DNA that does not code for transcription fation). They can be located up to 1 Mbp (1,000,000 bp) away from the gene, upstream or downstream from the start site [[2]](https://en.wikipedia.org/wiki/Enhancer_\(genetics\)).
- **Insulator**: An insulator is a genetic boundary element that blocks the interaction between enhancers and promoters. It has been found to cluster at the boundaries of topological association domains (TADs) and may have a role in partitioning the genome into "chromosome neighborhoods" - genomic regions within which regulation occurs [[3]](https://ipfs.io/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki/Insulator_\(genetics\).html).
- **Silencer**: A silencer is a DNA sequence capable of binding transcription regulation factors that block the binding of RNA polymerase to DNA sequence, thus provent genes from being expressed as proteins [[4]](https://en.wikipedia.org/wiki/Silencer_\(genetics\)).

![](/assets/promoter.png)
Figure1. Schematic overview of elements in eukaryotes.

##1.2 Multi-scale folding
The largest chromosomes contain hundreds of millions of base pairs that fold in a limitted space, which leads to multi-scale, hierarchical structures like： nucleosomes, chromatin fibres, chromosome domains, compartments and finally in chromosome territories. 

Information resides at all levels, from the histone–DNA interactions at the sub-nucleosomal scale to the chromosome–chromosome and chromosome–lamina interactions in the nuclear space. This multi-level architecture can be regulated and/or exploited by a variety of com- ponents such as transcription factors, architectural proteins and non-coding RNAs in order to coordinate gene expression and cell fate.

With the help of currently developed chromosome capture technologies, let's how them expanded our knowledge on chromosome structure.

### Chromosome territories
### A/B Compartments 
### TAD(Topologically Associating Domains)
### Sub TAD and insulation neighborhoods
### chromatin loops
### Nucleosome-nucleosome interactions

<br>

## 1.3 architectural proteins
- mediator
- cohesin
- CTCF

## 1.4 non-coding RNAs binding





**Referrence **
https://doi:10.1016/j.cell.2013.09.011 and https://www.youtube.com/watch?v=sAkH51R0DNg 1.0 <br>
https://doi.org/10.1016/j.tibs.2018.03.006 1.1<br>
http://dx.doi.org/10.1038/nrg.2016.112 1.1 1.2 1.3<br>
http://www.annualreviews.org/doi/10.1146/annurev-cellbio-100616-060531 1.1<br>
https://doi.org/10.1016/j.molcel.2013.02.011 1.1 structure<br>
http://www.genesdev.org/cgi/doi/10.1101/gad.281964.116. 1.1<br>




