# 4D project

## What is 4D Nucleome
To determine how the genome operates, we need to understand not only the linear encoding of information along chromosomes, but also its **three-dimensional organization and its dynamics across time**, that is, the ‘4D nucleome’.

## Current dilemma in understanding 4D nucleome
- Different experimental cell systems and approaches are used that together with the absence of **shared benchmarks** for assay performance have led to observations that cannot be directly compared. 
- Iimited ability to **integrate** different data types (for example, chromatin interaction data and imaging-based distance measurements)。
- Lack of approaches that can measure and account for **cell-to-cell variability** in chromosome and nuclear organization.
- Lack of **mechanistic insights** into the relationships between chromosome conformation and nuclear processes, including transcription, DNA replication and chromosome segregation.

## Goals and Strategies 
### Goals:
- Develop, benchmark, validate and standardize a wide array of **technologies to analyse** the 4D nucleome
- **Integrate**, analyse and model datasets obtained with these technologies to obtain a comprehensive view of the 4D nucleome
- Investigate the **functional role** of various structural features of chromosome organization in transcription, DNA replication and other nuclear processes. 

###Strategies:
- A set of **common cell lines** will be studied to enable direct cross- validation of data that are obtained with different methods [table](https://www.nature.com/articles/nature23884#t1).
- **Standards for data formats** and quality will be established so that data can be shared broadly.
- **Computational and analytical tools** will be developed to analyse individual datasets and to integrate, compare and cross-validate data obtained with different technologies.
- Develope multiple ways to measure **how DNA and trans-acting factors influence** on the 4D nucleosome.
- Develop a **consistent terminology** to describe nuclear features and biophysically derived principles guiding chromosome folding.
- **Compare and integrate** the wealth of information that is anticipated to be generated by the Network.
- Build a shared **database** and a public 4DN data browser includes all data, detailed protocols, engineered cell lines and reagents used across the Network.

## <a href="https://commonfund.nih.gov/4Dnucleome">Structure of the 4DN Network</a>
<table>
 <tr>
     <th>Consortium</th>
     <th>Institution Name</th>
     <th>Objections</th>
 </tr>
 <tr>
      <td><b>NOFIC<br>(Nuclear Organization<br> and Function<br>Interdisciplinary<br>Consortium) </td>
      <td><li>UNIVERSITY OF SOUTHERN CALIFORNIA</li>
        <li>UNIV OF ILLINOIS URBANA-CHAMPAIGN</li>
        <li>UNIV OF MASSACHUSETTS MED SCH WORCESTER</li>
        <li>LUDWIG INSTITUTE FOR CANCER RESEARCH LTD</li>
        <li>JACKSON LABORATORY</li>
        <li>UNIV OF WASHINGTON</li>
</td>
      <td><li>benchmark tools</li><li>Find best approach</li><li>Integrate approaches</li><li>Draft model of 4DN</li></td>
 </tr>
 <tr>
    <td><b>Imaging Tools</td>
    <td><li>YALE UNIVERSITY</li>
        <li>CALIFORNIA INSTITUTE OF TECHNOLOGY</li>
        <li>PRINCETON UNIVERSITY</li>
        <li>EUROPEAN MOLECULAR BIOLOGY LABORATORY</li>
        <li>UNIV OF MASSACHUSETTS MED SCH WORCESTER</li>
        <li>UNIVERSITY OF CALIFORNIA AT DAVIS</li>
        <li>STANFORD UNIVERSITY</li>
        <li>SALK INSTITUTE FOR BIOLOGICAL STUDIES</li>
        <li>ALBERT EINSTEIN COLLEGE OF MEDICINE</li>
    </td>
    <td><li>High res method</li><li>High-throughput</li><li>dynamic imaging</li><li>Live cell</li></td>
</tr>
 <tr>
    <td><b>DCIC<br>(Network Data<br>Coordination<br> and Integration<br> Center)</td>
    <td><li>HARVARD MEDICAL SCHOOL</li>
        <li>WASHINGTON UNIVERSITY</li>
    </td>
    <td><li>collect, store, curate, display and analyze data</li><li>develope efficient submission pipelines</li><li>high-quality, well- annotated, public accessible</li><li>user-friendly data portal</li></td>
</tr>
 <tr>
    <td><b>Network <br>Organizational <br>Hub</td>
    <td><li>UNIVERSITY OF CALIFORNIA SAN DIEGO</li>
    </td>
    <td><li>web portal for 4DN</li><li>integrated, bersatile data management
</li><li>retrieval, analysis and vis system</li><li>train and update 4DN members and collaboraters </li></td>
</tr>
 <tr>
    <td><b>Nuclear Bodies and Compartments</td>
    <td><li>PRINCETON UNIVERSITY</li>
    <li>SCRIPPS RESEARCH INSTITUTE</li>
    <li>PRINCETON UNIVERSITY</li>
    <li>FRED HUTCHINSON CANCER RES CENTER</li>
    <li>CALIFORNIA INSTITUTE OF TECHNOLOGY</li>
    <li>UNIV OF MASSACHUSETTS MED SCH WORCESTER</li>
    <li>COLUMBIA UNIVERSITY HEALTH SCIENCES</li>
    </td>
    <td><li>deciphering function of compartments</li><li>nucleoplasmic phase transitions</li><li>reversibly disrupt nuclear bodies</li></td>
    </tr>
 <tr>
    <td><b>Nucleomics Tools</td>
    <td><li>THE BABRAHAM INSTITUTE</li>
    <li>CALIFORNIA INSTITUTE OF TECHNOLOGY</li>
    <li>BAYLOR COLLEGE OF MEDICINE</li>
    <li>CORNELL UNIVERSITY</li>
    <li>UNIVERSITY OF PENNSYLVANIA</li>
    </td>
    <td><li>pipeline for single-cell hi-c</li><li>map dynamic organization</li><li>split pool barcoding</li><li>distance-Hi-C</li><li>super resolution vis</li></td>
    </tr>
</table>















