# 0.1 Molecular biology

[Molecular biology](https://en.wikipedia.org/wiki/Molecular_biology) concerns all molecular basis of a life from composition to activity, including the interactions between [DNA](https://en.wikipedia.org/wiki/DNA), [RNA](https://en.wikipedia.org/wiki/RNA), [proteins](https://en.wikipedia.org/wiki/Protein), their [biosynthesis](https://en.wikipedia.org/wiki/Protein_biosynthesis), as well as the regulation of these interactions.  Molecular biology also is the study of molecular underpinnings of the processes of [replication](https://en.wikipedia.org/wiki/DNA_replication), [transcription](https://en.wikipedia.org/wiki/Transcription_%28genetics%29), [translation](https://en.wikipedia.org/wiki/Translation_%28biology%29) and cell function, which is a good starting point to understand the field.

To better understand the content about 3D genome and serve a broad spectrum of readers, let's first review some important concepts. Folks who are familiar with these content feel free to skip this chapter. 

## Restriction enzyme

Restriction Enzymes that cuts DNA at specific nucleotide sequences known as restriction sites, were first discovered in bacteria where these enzymes are designed to **selectively** cut exogenous DNA \(like virus\) to protect themselves. 

To cut the DNA, restriction enzyme makes two incisions, each strand of the DNA double helix. These restriction sites are palindrome \( which means the  sequence read the same forwards as backwards \). There are over 3000 RE have been identified and more than 600 of them are commercially available. Details of enzyme can be found in [database](http://rebase.neb.com/rebase/rebase.html) or commercial website.  

Naturally occurring restriction endonucleases are categorized into four groups \(Types I, II III, and IV\) based on their composition and [enzyme cofactor](https://en.wikipedia.org/wiki/Enzyme_cofactor) requirements, the nature of their target sequence, and the position of their DNA cleavage site relative to the target sequence. Here we will focus on the last criteria.

<table>
  <thead>
    <tr>
      <th style="text-align:left"></th>
      <th style="text-align:center">Cleavage Site</th>
      <th style="text-align:center">Examples</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align:left">Type I</td>
      <td style="text-align:center">cut DNA at random far from their recognition sequences.</td>
      <td style="text-align:center">
        <p>EcoK I</p>
        <p>EcoA I</p>
        <p>CfrA I</p>
      </td>
    </tr>
    <tr>
      <td style="text-align:left">Type II</td>
      <td style="text-align:center">
        <p>Specific</p>
        <p>Within the recognition site</p>
      </td>
      <td style="text-align:center">
        <p>EcoR I</p>
        <p>BamH I</p>
        <p>Hind III</p>
      </td>
    </tr>
    <tr>
      <td style="text-align:left">Type III</td>
      <td style="text-align:center">
        <p>Random</p>
        <p>24-26 bp away from recognition site</p>
      </td>
      <td style="text-align:center">
        <p>EcoP I</p>
        <p>Hinf III</p>
        <p>EcoP15 I</p>
      </td>
    </tr>
  </tbody>
</table>## Hydrogen bonds

Hydrogen bond  is a partially [electrostatic](https://en.wikipedia.org/wiki/Electrostatics) attraction between a [hydrogen](https://en.wikipedia.org/wiki/Hydrogen) \(H\) which is bound to a more [electronegative](https://en.wikipedia.org/wiki/Electronegativity) atom such as [nitrogen](https://en.wikipedia.org/wiki/Nitrogen) \(N\), [oxygen](https://en.wikipedia.org/wiki/Oxygen)\(O\), or [fluorine](https://en.wikipedia.org/wiki/Fluorine) \(F\), and another adjacent atom bearing a [lone pair](https://en.wikipedia.org/wiki/Lone_pair) of electrons \(according to wiki\). In another word it is an attraction between a slightly positive hydrogen atom and a slightly negative atom.

All life depends on hydrogen bonds in basic form water. It also plays an important role in determining the three-dimensional structures and the properties adopted by many synthetic and natural proteins. DNA, proteins, cellulose and so on.

**In epigenetics**, there are 142 hydrogen bonds between DNA and the histone core in each nucleosome.  More than 1/5 of the amino acids in each of the core histones are either lysine or arginine, and their positive charges neutralize the negatively charged DNA backbone. 

For **protein-DNA recognition**,  the mechanism is based on base readout and shape readout. Base readout is when the protein recognizes the specific chemical signatures of different nucleic acid bases which extensively depends on hydrogen bonds. In protein-DNA recognition, it is a greater source of specificity in the major groove as compared to the minor groove due to the pattern of hydrogen bond donors and acceptors available [\[1\]](https://en.wikibooks.org/wiki/Structural_Biochemistry/Protein_function/DNA_Binding).

## Gel electrophoresis

[Gel electrophoresis](http://en.wikipedia.org/wiki/Gel_electrophoresis) is a method to separate [DNA](http://en.wikipedia.org/wiki/DNA), [RNA](http://en.wikipedia.org/wiki/RNA) or even proteins by size and charge. It is used in [biochemistry](https://en.wikipedia.org/wiki/Biochemistry) and [molecular biology](https://en.wikipedia.org/wiki/Molecular_biology) to separate a mixed population of DNA and RNA fragments by length, to estimate the size of DNA and RNA fragments or to separate proteins by charge.[\[2\]](https://en.wikipedia.org/wiki/Gel_electrophoresis#cite_note-1)

This is achieved by moving negatively charged nucleic acid molecules \(like DNA\) through an [agarose](http://en.wikipedia.org/wiki/Agarose) matrix with an [electric field](http://en.wikipedia.org/wiki/Electric_field) \([electrophoresis](http://en.wikipedia.org/wiki/Electrophoresis)\). The speed of molecular depends on various factors like: strength of the electrical field, buffer, density of agarose gel and so on, but most importantly is the size of DNA. Shorter molecules move faster and migrate farther than longer ones due to their mobility through pores in the gel. Within an agarose gel, linear DNA migrate inversely proportional to the log10 of their molecular weight.

Gel electrophoresis has many applications such like:

* Estimation of the size of DNA molecules after using restriction enzyme digestion, the size could be estimated by comparing with some standard reference markers \( whose length is known \). 
* Analysis of [PCR](https://en.wikipedia.org/wiki/PCR) products, e.g. in molecular [genetic diagnosis](https://en.wikipedia.org/wiki/Preimplantation_genetic_diagnosis) or [genetic fingerprinting](https://en.wikipedia.org/wiki/Genetic_fingerprinting).
* Separation of restricted genomic DNA by cutting down the block of gel contains specific sequence. 

## PCR

[Polymerase Chain Reaction](https://en.wikipedia.org/wiki/Polymerase_chain_reaction) is a method to amplify **a particular piece** of DNA, it can make billions of copies of a target sequence of DNA in a few hours. PCR was invented in the 1984 as a way to make numerous copies of DNA fragments in the laboratory, this technique has been applied in enormous various applications and really enriched the field of molecular biology. 

We can think PCR as an [in vitro](https://en.wikipedia.org/wiki/In_vitro) version of **DNA replication** in vivo. DNA replication is semi-conservative which means only one strand of the DNA is used as the template for the growth of a new DNA strand. To this end, we need few key enzymes to be involved:

* **DNA Polymerase:** Catalyzes the elongation of DNA by adding nucleoside triphosphates to the **3’ end** of the growing strand.
* **DNA Ligase:** Given the antiparallel nature of DNA and the fact that DNA ploymerases can only add nucleotides to the 3’ end, one strand \(referred to as the leading strand\) of DNA is synthesized continuously and the other strand \(referred to as the lagging strand\) in synthesized in fragments \(called [Okazaki fragments](https://en.wikipedia.org/wiki/Okazaki_fragments)\) that are joined together by DNA ligase.
* **Primase:**  In humans, primase is the enzyme that can start an RNA chain from scratch and it creates a primer \(a short RNA with an available 3’ end\).
* **Helicase:**  untwists the two parallel DNA strands.
* **Topoisomerase:**  relieves the stress of this twisting.
* **Single strand binding protein:**  binds to and stabilizes the unpaired DNA strands.

Besides all these enzymes, there are three main steps that finish the magic to replicate the desired sequence from one to millions:

* Step 1: Denature DNA

        At 95°C, the DNA is denatured \(i.e. the two strands are separated\).

* Step 2: Primers Anneal.

       At 40°C- 65°C, the primers anneal \(or bind to\) their complementary sequences on the single strands of DNA.

* Step 3: DNA polymerase Extends the DNA chain

       At 72°C, DNA Polymerase extends the DNA chain by adding nucleotides to the 3’ ends of the primers.

{% embed url="https://www.youtube.com/watch?v=nHi-3jP6Mvc" %}

Even though PCR is such an powerful tool, there are still some **caveats** we should notice and they may produce **biases** during many applications. Reviews about this topic can be found [\[3\]](https://academic.oup.com/nar/article/43/21/e143/2468099), [\[4\], ](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1317340/)[\[5\]](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3188800/).

