# 3.1 Assessment of Primary-order Chromatin
Genomic regions with dense nucleo- somes are more tightly packed (i.e., “closed”), whereas nucleosome- depleted regions are more accessible (i.e., “open”) for interactions with regulators and are therefore regarded as the primary locations ofregulatory elements. 

## 3.2.1 Features of data
![](/assets/prios.jpg)
**Figure1.Fragmentation methods and read out features comparison between four methods.**MNase is specially sensitive to open chromotin areas and it is exonucleas which maps regions that are protected by nucleosomes (large open regions are digested). DNase-seq and ATAC-seq are used to sequence and map exposed regions of DNA. 
## 3.2.2 Analytical pipeline:
### Step1 Raw reads (Quality control)

 - Composite plots to check experiment sucess: eg, TSSs shown t be open.---- **ArchTEX[[1]](https://www.ncbi.nlm.nih.gov/pubmed/22302569), CEAS[[2]](https://www.ncbi.nlm.nih.gov/pubmed/19689956)**.
 - ATAC-seq can be further: estimating the percentage of sequence reads that map to the mitochondrial genome(lower is better).
 - Use genome browser to see raw tag density ---- **UCSC[[3]](https://www.ncbi.nlm.nih.gov/pubmed/20959295), IGV[[4]](https://www.ncbi.nlm.nih.gov/pubmed/22517427), [GIVE](https://www.biorxiv.org/content/early/2018/03/15/177832)**.
 
### Step2 Pre-processing & Alignment 
Reads are filtered to remove redundancy and adaptors, also size selection (selective for some methods).
#### Size selection:
- MNase: 25-50bp fragments represent TF-binding sites.
- ATAC: 
 - reads length below **38bp** are removed,as 38 bp is the minimal distance between neighboring transposition sites generated by the Tn5 transposase.
 - mitochondrial genome reads removed.

#### Alignment 
- **Full-read approach**
 - Bowtie2
 - Burrows-Wheeler Aligner (BWA)
- **Chimeric alignment** (unmapped from previous step, which may contains ligation juction)
 - BWA
 - STAR
 - ChimeraScan
 
#### Minimal reads number
 - MNase: 150-200 Million reads.
 - DNase / FAIRE: 25-50 M
 - ATAC:  50-160 M
 
 
### Step3 Peak calling 
Peak calling basically means we identify possible open regions or nucleosome location through the tiling reads depth. This step is the most critical for chromatin accessibility profiling, revealing nucleosome-dense, closed regions (MNase-seq) or open chromatin regions (DNase-seq, FAIRE-seq, and ATAC-seq) [[5]](https://doi.org/10.1016/j.csbj.2018.02.003).

#### For MNase-seq
##### Midpoint of nucleosome:
- **single-end mode**: shifting the ends 73 bp from 5' end, or extending the ends from 120 to 147 bp in the 3′ direction. 
- **paired end mode**: mid of forward and reverse reads. 

##### Peak calling tools:
<table>
 <tbody>
  <tr>
    <th>Tools</th>
    <th>Description</th>
    <th>Reference</th>
  </tr>
  <tr>
    <td>GeneTrack</td>
    <td><ul><li>Gaussian smoothing</li><li>generate prob-based map based on predefined exclusion distance</li></ul></td><td>[6]</td>
  </tr>
  <tr>
    <td>iNPS</td>
    <td><ul><li>wave-like structure of data</li><li>first derivative of the Gaussian smoothed profile</li></ul></td><td>[7]</td>
   </tr>  
   <tr>
      <td>DANPOS</td>
      <td><ul><li>identification ofdynamic nucleosomes positions</li></ul></td><td>[8]</td>
      </tr>
</tbody>
</table>

#### For DNase-seq and FAIRE-seq
##### Specific tools for DNase-seq
<table>
 <tbody>
  <tr>
    <th>Tools</th>
    <th>Description</th>
    <th>Reference</th>
  </tr>
  <tr>
    <td>F-seq</td>
    <td><ul><li>smooth Gaussian kernel density estimation</li><li>higher accuracy and sensitivity</li></ul></td><td>[9]</td>
  </tr>
  <tr>
    <td>Hotspot</td>
    <td><ul><li>reports statistical signifi- cance for identified DHSs<br>(DNase I hypersensitive site)</li></ul></td><td>[10]</td>
   </tr>  
</tbody>
</table>
##### Specific tools for FAIRE-seq
<table>
 <tbody>
  <tr>
    <th>Tools</th>
    <th>Description</th>
    <th>Reference</th>
  </tr>
  <tr>
    <td>MACS2</td>
    <td><ul><li>more parameter settings</li><li>settings can be converted to P-values empirically</li></ul></td><td>[11]</td>
  </tr>
</tbody>
</table>

##### General peak calling tools
<table>
 <tbody>
  <tr>
    <th>Tools</th>
    <th>Description</th>
    <th>Reference</th>
  </tr>
  <tr>
    <td>ZINBA</td>
    <td><ul><li>regression model to identify enriched regions without a prior</li><li>regions within a defined dis- tance are combined to form a broad region</li><li>shape-detection function for sharp signals</li><li>can combine prior like GC content when noise is high</li></ul></td><td>[12]</td>
  </tr>
  <tr>
    <td>MACS</td>
    <td><ul><li>use Poisson distribution as a background model</li></ul></td><td>[11]</td>
   </tr>  
</tbody>
</table>




#### For ATAC-seq
Paired-end sequencing is performed for ATAC-seq. The read start sites require **adjustment** because the Tn5 transposase binds as a dimer and inserts adaptors separated by 9bp. Generally, reads aligned to the + strand are offset by +4 bp, and reads aligning to the - strand are offset by −5 bp [[5]](https://doi.org/10.1016/j.csbj.2018.02.003). This method can detect both broad regions (few kilobase) or small regions (50-500bp).
##### Comman peak calling tools:
MACS2, ZINBA, F-seq, HOMER. ["atac-seq"-R package](https://github.com/ blikzen/atac-seq).

### Step4 Chromatin Accessibility Analysis
Accessible regions are determined based on peak-calling results. With peak files we go downstream analysis with different purpose:
<table>
 <tbody>
  <tr>
    <th></th>
    <th>MNase-seq</th>
    <th>DNase-seq</th>
    <th>FAIRE-seq</th>
    <th>ATAC-seq</th>
  </tr>
  <tr>
    <th>Identification<br>Objectives</th>
    <td>Nucleosome positioning</td><td>TF footprint</td><td>CORE locations</td><td>Nucleosome pos and TF footprint</td>
  </tr> 
  <tr>
    <th>Tools</th>
    <td><ul><li>Nucleosome positioning algorithms [13-16]</li><li>Nucleosome occupancy algorithms [17-18]</li><li>V-plots for TF occupancy [19]</li></ul></td><td><ul><li>Digital genomic footprinting algorithms [20-21]</li><li>Nucleosome and TF occupancy algorithms [22]</li><li>CENTIPEDE [23]</li><ul></td>
    <td>Not available</td>
    <td><ul><li>Digital genomic footprinting algorithms</li><li>CENTIPEDE</li></ul></td>
  </tr>
</tbody>
</table>
**V-plots** are two dimensional dot- plots that display each fragment’slengthinthe Y-axis versus the corresponding fragment midpoint position in the X-axis for TF binding detection.
<br>
**CENTIPEDE** is an integrative algorithm for rapid profiling of many TFs simultaneously that combines known information on TF motifs and positional weight matrices
 
 
### Step5 Analysis and Interpretation
Data annotation and integration represents the final and most informative stage of analysis. After we get the nucleosome positions and TF binding sites, it is desireble to further interpret them in the light of relavent information.
- In combination with (promoters, introns, intergenic regions, TSSs, TTSs) information ---- [BedTools](http://bedtools.readthedocs.org/).
- Discovery TF binding events based on known knowledge or *de novo* ways.

## 3.2.3 Analytical Tools for peak calling:
### Brief view of different tools 
A comprehensive list for peak calling softwares can be found at [here](https://omictools.com/peak-calling-category). Few widely-used tools are listed below. 
<table>
  <tbody>
   <tr>
    <td>MACS2 (MACS1.4)</td> <td>Most widely used peak caller. Can detect narrow and broad peaks.</td>
  </tr>
   <tr>
    <td>Epic (SICER)</td>
    <td>Specialised for broad peaks</td>
  </tr>
   <tr>
    <td>BayesPeak R/Bioconductor</td>
    <td>Jmosaics Detects enriched regions jointly from replicates</td>
  </tr>
   <tr>
    <td>EDD</td>
    <td>Detects megabase domain enrichment</td>
  </tr>
   <tr>
    <td>GEM</td><td>Peak calling and motif discovery for ChIP-seq and ChIP-exo</td>
  </tr>
  <tr>
    <td>SPP</td><td>Fragment length computation and saturation analysis to determine if read depth is</td>
  </tr>
  </tbody>
<table>

### Introduction of MACS
MACS is on of the most popular peak calling tools and is developed in  X. Shirley Liu’s lab at Harvard University. We'll use MACS as an example to show you the basic idea behind peak calling and practical implementation. The original paper introduce MACS can be found [here](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2008-9-9-r137).

#### Core idea of the algorithm
The combination of TF and genome is a relatively random process, that's to say that every position on the genome have the chance to be seen by the TF (however, with different probablity). Peak calling is aimed to find those hot spot that is easily seen.

How can we identify whether a spot is "hot"? Suppose we are sequencing a group of cells, then a hot spot is presumably be covered more frequently than other spots. This process can be viewed as a binomial distribution and    when the number of reads *n* is large and the prob *p* of each location to be covered by a read is small, then the binomial distribution became [Poisson distribution](https://en.wikipedia.org/wiki/Poisson_distribution). 
> An event can occur 0, 1, 2, … times in an interval. The average number of events in an interval is designated <math> \lambda </math> (lambda). Lambda is the event rate, also called the rate parameter. The probability of observing {{mvar|k}} events in an interval is given by the equation:    <math>P(k \text{ events in interval}) = e^{-\lambda}\frac{\lambda^k}{k!}</math>

Here the <math> \lambda=n*p, p=\frac{l}{s}</math>. 
- n: the reads number from sequencing.
- l: length of a single read.
- s: length of the whole genome.

With this basic model then we can compute a threshold depth of a spot by setting a confidence probability. Howver, this is only a baby model which may fail in real circumstances due to lots of bias of the experiment. 

#### Bias in ChIP-seq experiment
In doing ChIP-seq analysis, it's important to think about biases that can affect your results. There are a number of issues, such as issues with chromatin accessibility that are going to affect how your DNA gets fragmented, issues with amplification, repetitive regions, which are going to be difficult to map back to. And so it's very important in a ChIP-seq experiment to use some kind of control, and a common control is to use input DNA control, where we have data that's, that is fragmented, but the immunoprecipitation, or the, the antibody pull down has not been performed. (A comprehensive review for bias can be found [here](http://dx.doi.org/10.1038/nrg3788)).

#### Using control data
Determining the background expectation of the number of peaks you would see and use that in making your peak calls. It also allows for MACS to compute a false discovery rate. Without a control, MACS will not compute a false discovery rate. But using a control, it's able to to model that background. 

#### Generate profile from combined tag
Another problem is that, the reads we got do not indicate the real potion of a TF, reads are pulled down together with a TF. In that case we usually get a double peak result which desired further analysis. MACS solved this problem by shifting tags of d/2, where d  is the discrepancy between forward and backward peaks around one TF. An illustration can be seen [here](https://www.nature.com/articles/nrg2641/figures/5).

#### Implementation Pipeline
Intall [MACS](http://liulab.dfci.harvard.edu/MACS/Download.html)(version 1.4 or 2.0).

```
tar xvzf MACS-1.4.2-1.tar.gz
cd MACS-1.4.2
python setup.py install –prefix /your_directory/
```

Change the enviroment path

```
export PATH = /your_directory/bin:$PATH
export PYTHONPATH = /your_directory/lib/python2.X/site-packages/:$PYTHONPATH
```

Align the reads to the genome:

```
bowtie –m 1 -S -q /path_to/mm10 CTCF.fastq CTCF.sam # details refer to usage of bowtie
```

Run MACS:

```
macs14 -t CTCF.sam -c Control.sam -n CTCF –g mm 
```

The help docs of comprehensive parameters [here](http://liulab.dfci.harvard.edu/MACS/README.html).
After alignment we'll get four files: **CTCF_model.r，CTCF_peaks.bed，CTCF_peaks.xls，CTCF_summits.bed**. The double peaks figure is:
```
 Rscript CTCF_model.r
```
The CTCF_peak.bed records the location of each peak and can be futher processed with the tools like Bedtools.


# Referrence
[1] Lai, William KM, Jonathan E. Bard, and Michael J. Buck. "ArchTEx: accurate extraction and visualization of next-generation sequence data." Bioinformatics 28.7 (2012): 1021-1023.<br>
[2] Shin, Hyunjin, et al. "CEAS: cis-regulatory element annotation system." Bioinformatics 25.19 (2009): 2605-2606.<br>
[3] Fujita, Pauline A., et al. "The UCSC genome browser database: update 2011." Nucleic acids research 39.suppl_1 (2010): D876-D882.<br>
[4] Thorvaldsdóttir, Helga, James T. Robinson, and Jill P. Mesirov. "Integrative Genomics Viewer (IGV): high-performance genomics data visualization and exploration." Briefings in bioinformatics 14.2 (2013): 178-192.<br>
[5] Chang, Pearl, et al. "Computational Methods for Assessing Chromatin Hierarchy." Computational & Structural Biotechnology Journal (2018).<br>
[6] Albert I, Wachi S, Jiang C, Pugh BF. GeneTrack - a genomic data processing and visualization framework. Bioinformatics 2008;24(10):1305–6.<br>
[7] Chen W, Liu Y, Zhu S, Green C, Wei G, Han JD. Improved nucleosome-positioning algorithm iNPS for accurate nucleosome positioning from sequencing data. Nat Commun 2014;5:4909.<br>
[8] ChenK, XiY, Pan X, Li Z, Kaestner K, Tyler J, et al. DANPOS: dynamic analysis ofnucle- osome position and occupancy by sequencing. Genome Res 2013;23(2):341–51.<br>
[9] Boyle AP, Guinney J, Crawford GE, Furey TS. F-Seq: a feature density estimator for high-throughput sequence tags. Bioinformatics 2008;24(21):2537–8.<br>
[10] Baek S, Sung MH, Hager GL. Quantitative analysis of genome-wide chromatin remodeling. Methods Mol Biol 2012;833:433–41.<br>
[11] Zhang Y, Liu T, Meyer CA, Eeckhoute J, Johnson DS, Bernstein BE, Nusbaum C, Myers RM, Brown M, Li W, Liu XS: Model-based analysis of ChIP-Seq (MACS). Genome Biol 2008, 9(9):R137<br>
[12] Rashid NU, Giresi PG, Ibrahim JG, Sun W, Lieb JD. ZINBA integrates local covariates with DNA-seq data to identify broad and narrow regions of enrichment, even within amplified genomic regions. Genome Biol 2011;12(7):R67.<br>
[13] Kaplan N, Moore IK, Fondufe-Mittendorf Y, Gossett AJ, Tillo D, Field Y, LeProust EM, Hughes TR, Lieb JD, Widom J, Segal E: The DNA-encoded nucleosome organization of a eukaryotic genome. Nature 2009, 458(7236):362–366.<br>
[14] Weiner A, Hughes A, Yassour M, Rando OJ, Friedman N: High-resolution nucleosome mapping reveals transcription-dependent promoter packaging. Genome Res 2010, 20(1):90–100.<br>

[15] Albert I, Mavrich TN, Tomsho LP, Qi J, Zanton SJ, Schuster SC, Pugh BF: Translational and rotational settings of H2A.Z nucleosomes across the Saccharomyces cerevisiae genome. Nature 2007, 446(7135):572–576.<br>

[16] Zhang Y, Moqtaderi Z, Rattner BP, Euskirchen G, Snyder M, Kadonaga JT, Liu XS, Struhl K: Intrinsic histone-DNA interactions are not the major determinant of nucleosome positions in vivo. Nat Struct Mol Biol 2009, 16(8):847–852.<br>

[17] Kaplan N, Moore IK, Fondufe-Mittendorf Y, Gossett AJ, Tillo D, Field Y, LeProust EM, Hughes TR, Lieb JD, Widom J, Segal E: The DNA-encoded nucleosome organization of a eukaryotic genome. Nature 2009, 458(7236):362–366<br>

[18] Mavrich TN, Ioshikhes IP, Venters BJ, Jiang C, Tomsho LP, Qi J, Schuster SC, Albert I, Pugh BF: A barrier nucleosome model for statistical positioning of nucleosomes throughout the yeast genome. Genome Res 2008, 18(7):1073–1083.<br>

[19] Henikoff JG, Belsky JA, Krassovsky K, MacAlpine DM, Henikoff S: Epigenome characterization at single base-pair resolution. Proc Natl Acad Sci U S A 2011, 108(45):18318–18323.<br>
[20] Boyle AP, Song L, Lee BK, London D, Keefe D, Birney E, Iyer VR, Crawford GE, Furey TS: High-resolution genome-wide in vivo footprinting of diverse transcription factors in human cells. Genome Res 2011, 21(3):456–464.<br>
[21] Neph S, Vierstra J, Stergachis AB, Reynolds AP, Haugen E, Vernot B, Thurman RE, John S, Sandstrom R, Johnson AK, Maurano MT, Humbert R, Rynes E, Wang H, Vong S, Lee K, Bates D, Diegel M, Roach V, Dunn D, Neri J, Schafer A, Hansen RS, Kutyavin T, Giste E, Weaver M, Canfield T, Sabo P, Zhang M, Balasundaram G, et al: An expansive human regulatory lexicon encoded in transcription factor footprints. Nature 2012, 489(7414):83–90<br>
[22] Vierstra J, Wang H, John S, Sandstrom R, Stamatoyannopoulos JA: Coupling transcription factor occupancy to nucleosome architecture with DNase-FLASH. Nat Methods 2014, 11(1):66–72.<br>
[23] Pique-Regi R, Degner JF, Pai AA, Gaffney DJ, Gilad Y, Pritchard JK: Accurate inference of transcription factor binding from DNA sequence and chromatin accessibility data. Genome Res 2011, 21(3):447–455.<br>



