# 3.1 Assessment of Primary-order Chromatin
Genomic regions with dense nucleo- somes are more tightly packed (i.e., “closed”), whereas nucleosome- depleted regions are more accessible (i.e., “open”) for interactions with regulators and are therefore regarded as the primary locations ofregulatory elements. 

## 3.2.1 Features of data
![](/assets/prios.jpg)
**Figure1.Fragmentation methods and read out features comparison between four methods.**MNase is specially sensitive to open chromotin areas and it is exonucleas which maps regions that are protected by nucleosomes (large open regions are digested). DNase-seq and ATAC-seq are used to sequence and map exposed regions of DNA. 
## 3.2.2 Analytical pipeline:
### Step1 Raw reads (Quality control)

 - Composite plots to check experiment sucess: eg, TSSs shown t be open.----**ArchTEX[108], CEAS[110]**.
 - ATAC-seq can be further: estimating the percentage of sequence reads that map to the mitochondrial genome(lower is better).
 - Use genome browser to see raw tag density ---- **UCSC[121],IGV[122],GIVE**.
 
### Step2 Pre-processing & Alignment 
Reads are filtered to remove redundancy and adaptors, also size selection (selective for some methods).
#### Size selection:
- MNase: 25-50bp fragments represent TF-binding sites.
- ATAC: 
 - reads length below **38bp** are removed,as 38 bp is the minimal distance between neighboring transposition sites generated by the Tn5 transposase[2'].
 - mitochondrial genome reads removed.

#### Alignment 
- **Full-read approach**
 - Bowtie2[57']
 - Burrows-Wheeler Aligner (BWA)[58']
- **Chimeric alignment** (unmapped from previous step, which may contains ligation juction)
 - BWA
 - STAR
 - ChimeraScan【？】
 
#### Minimal reads number 【引用？！】
 - MNase: 150-200 Million reads.
 - DNase / FAIRE: 25-50 M
 - ATAC:  50-160 M
 
 
### Step3 Peak calling 
Peak calling basically means we identify possible open regions or nucleosome location through the tiling reads depth. This step is the most critical for chromatin accessibility profiling, revealing nucleosome-dense, closed regions (MNase-seq) or open chromatin regions (DNase-seq, FAIRE-seq, and ATAC-seq)【'论文本身】.

#### For MNase-seq
##### Midpoint of nucleosome:
- **single-end mode**: shifting the ends 73 bp from 5' end, or extending the ends from 120 to 147 bp in the 3′ direction. 
- **paired end mode**: mid of forward and reverse reads. 

##### Peak calling tools:
<table>
 <tbody>
  <tr>
    <th>Tools</th>
    <th>Description</th>
    <th>Reference</th>
  </tr>
  <tr>
    <td>GeneTrack</td>
    <td><ul><li>Gaussian smoothing</li><li>generate prob-based map based on predefined exclusion distance</li></ul></td><td>[1-4】</td>
  </tr>
  <tr>
    <td>iNPS</td>
    <td><ul><li>wave-like structure of data</li><li>first derivative of the Gaussian smoothed profile</li></ul></td><td>[1-20】</td>
   </tr>  
   <tr>
      <td>DANPOS</td>
      <td><ul><li>identification ofdynamic nucleosomes positions</li></ul></td><td>[1-19】</td>
      </tr>
</tbody>
</table>

#### For DNase-seq and FAIRE-seq
##### Specific tools for DNase-seq
<table>
 <tbody>
  <tr>
    <th>Tools</th>
    <th>Description</th>
    <th>Reference</th>
  </tr>
  <tr>
    <td>F-seq</td>
    <td><ul><li>smooth Gaussian kernel density estimation</li><li>higher accuracy and sensitivity[2-135】</li></ul></td><td>[1-15】</td>
  </tr>
  <tr>
    <td>Hotspot</td>
    <td><ul><li>reports statistical signifi- cance for identified DHSs<br>(DNase I hypersensitive site)</li></ul></td><td>[1-8,49】</td>
   </tr>  
</tbody>
</table>
##### Specific tools for FAIRE-seq
<table>
 <tbody>
  <tr>
    <th>Tools</th>
    <th>Description</th>
    <th>Reference</th>
  </tr>
  <tr>
    <td>MACS2</td>
    <td><ul><li>more parameter settings</li><li>settings can be converted to P-values empirically</li></ul></td><td>[1-132】</td>
  </tr>
</tbody>
</table>

##### General peak calling tools
<table>
 <tbody>
  <tr>
    <th>Tools</th>
    <th>Description</th>
    <th>Reference</th>
  </tr>
  <tr>
    <td>ZINBA</td>
    <td><ul><li>regression model to identify enriched regions without a prior</li><li>regions within a defined dis- tance are combined to form a broad region</li><li>shape-detection function for sharp signals</li><li>can combine prior like GC content when noise is high</li></ul></td><td>[1-72】</td>
  </tr>
  <tr>
    <td>MACS</td>
    <td><ul><li>use Poisson distribution as a background model</li></ul></td><td>[1-8,49】</td>
   </tr>  
</tbody>
</table>




#### For ATAC-seq
Paired-end sequencing is performed for ATAC-seq. The read start sites require **adjustment** because the Tn5 transposase binds as a dimer and inserts adaptors separated by 9bp. Generally, reads aligned to the + strand are offset by +4 bp, and reads aligning to the - strand are offset by −5 bp【文章1】. This method can detect both broad regions (few kilobase) or small regions (50-500bp).
##### Comman peak calling tools:
MACS2, ZINBA, F-seq, HOMER. ["atac-seq"-R package](https://github.com/ blikzen/atac-seq).

### Step4 Chromatin Accessibility Analysis
Accessible regions are determined based on peak-calling results. With peak files we go downstream analysis with different purpose:
<table>
 <tbody>
  <tr>
    <th></th>
    <th>MNase-seq</th>
    <th>DNase-seq</th>
    <th>FAIRE-seq</th>
    <th>ATAC-seq</th>
  </tr>
  <tr>
    <th>Objectives</th>
    <td>Nucleosome positioning</td><td>TF footprint</td><td>CORE locations</td><td>Nucleosome pos and TF footprint</td>
  </tr> 
  <tr>
    <th>Tools</th>
    <td><ul><li>Nucleosome positioning algorithms [2-48,58,111,144】</li><li>Nucleosome occupancy algorithms [2-48,145】</li><li>V-plots for TF occupancy [2-50】</li></ul></td><td><ul><li>Digital genomic footprinting algorithms [19,78,83,85,128,146-149】</li><li>Nucleosome and TF occupancy algorithms [2-150】</li><li>CENTIPEDE [2-151】</li><ul></td>
    <td>Not available</td>
    <td><ul><li>Digital genomic footprinting algorithms</li><li>CENTIPEDE</li></ul></td>
  </tr>
</tbody>
</table>





 
 
### Step5 Analysis and Interpretation


## 3.2.3 Analytical Tools:
- Brief view of different tools 
- Give suggestions on how to choose proper tools 

**Referrence **
https://doi.org/10.1016/j.csbj.2018.02.003 *



