##2.2- C-Techs (chromosome conformation capture)-coupled 

<table>
<tbody>
<tr>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Rectangular-heatmap"><img src="https://cloud.githubusercontent.com/assets/2143629/24176191/e66e26e0-0e70-11e7-84fd-e945b95048b7.png" height="50px"></a></td>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Rectangular-heatmap">Rectangular heatmap</a></td>
<td><ul><li>Transforms(ICE,KR,etc)-see Chap3.2</li><li>Value scaling(linear,log)</li><li>Zoom limit(5k-20M)</li></ul></td>
</tr>
<tr>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Horizontal-and-vertical-heatmaps"><img src="https://cloud.githubusercontent.com/assets/2143629/24176286/67e87978-0e71-11e7-9441-cdabbf2fc804.png" height="50px"></a></td>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Horizontal-and-vertical-heatmaps">Horizontal and vertical heatmaps</a></td>
<td><ul><li>45° roatate of 2D heatmap</li><li>Can be added lef,right,top,bottom</li><li>Default faces: down & right(can be fpliiped)</li></ul></td>
</tr>
<tr>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Line-tracks"><img src="https://cloud.githubusercontent.com/assets/2143629/24176389/0acb6b8c-0e72-11e7-9396-5e0e7a242d82.png" height="50px"></a></td>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Line-tracks">Line</a></td>
<td><ul><li>Scaling(linear,log)</li><li>Label position</li><li>Axis position</li><li>Stroke color</li></ul></td>
</tr>
<tr>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Chromosome-grid"><img src="https://cloud.githubusercontent.com/assets/2143629/24176445/677221f0-0e72-11e7-81f3-ae29c71c1f6d.png" height="50px"></a></td>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Chromosome-grid">Chromosome grid</a></td>
<td><ul><li>Grid on chromosome scale</li><li>User can define the scale</li></ul></td>
</tr>
<tr>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Chromosome-axis"><img src="https://cloud.githubusercontent.com/assets/2143629/24176330/b38cb0c4-0e71-11e7-9ed7-60e8bee1f726.png" height="50px"></a></td>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Chromosome-axis">Chromosome axis</a></td>
<td><ul><li>Diff scale for zoom in&out</li><li>User can define the scale</li></ul></td>
</tr>
<tr>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Viewport-projection"><img src="https://cloud.githubusercontent.com/assets/2143629/24176502/afeb5f46-0e72-11e7-8c1d-0c3d12d75eff.png" height="50px"></a></td>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Viewport-projection">Viewport projection</a></td>
<td><ul><li>show one dataset with two different resolutions</li><li>Open from view</li></ul></td>
</tr>
<tr>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Gene-annotations"><img src="https://cloud.githubusercontent.com/assets/2143629/24176594/48af7d98-0e73-11e7-8a10-81c5e64b41be.png" height="50px"></a></td>
<td><a href="https://github.com/hms-dbmi/higlass/wiki/Gene-annotations">Gene annotations</a></td>
<td><a href="https://github.com/hms-dbmi/clodius/wiki/Gene-annotations">Developer docs</a></td>
</tr>
</tbody>
</table>






inferred from pairwise contact frequencies, proximity ligation. 
(expalin what one and many mean, pros and cons. )
(draw a illustration try to cover the main experimental difference between diff methods)
[1][2] 
### 2.2.1 One-to-one:
- 3C

### 2.2.2 One-to-all:
- Multiplexed 3C-seq
- Open-ended 3C
- 3C-DSL
- 4C
- 4C-seq
- TLA
- e4C
- ACT

### 2.2.3 Many-to-many:
- 5C
- ChIA-PET 

### 2.2.4 Many-to-all
- Capture-3C
- Capture-HiC
- PLAC-seq/HiChIP

### 2.2.5 All to all 
- GCC
- Hi-C
- ELP
- TCC

### Hi-C enrichment
- Single-cell
- In situ/DNase/Micro-HiC

## Another methods for classification:
### Locus-landmark: 
Measure interactions of genomic loci with relatively fixed nuclear ‘landmarks’
- Techniques: ChIP, DamID, TSA-seq

### Locus-locus: 
Measure interactions between genomic loci
- Techniques: 3C, Hi-C, ChIA-PET etc.



**Referrence **
https://doi.org/10.1016/j.csbj.2018.02.003<br>
https://github.com/hms-dbmi/hic-data-analysis-bootcamp/blob/master/HiC-Protocol.pptx 


