##2.2- C-Techs (chromosome conformation capture)-coupled 

The foundamental object of 3C(Chromosome Conformation Capture) techniques and its derivation is to understand the physical wiring diagram of the genome by identifying the physical interaction between chromosomes. 

To capture the interaction ( crosslink between strings), there are few steps to follow in general:
- Take a snapshot of the flowing cells - **Crosslink** with fixative agent (formaldehyde)
- Zoom in on crosslinked part and exclude untangled parts - **Digested** with a restriction enzyme
- Analyze the components come from the same chromatin - **Reverse crosslink** and **sequence**







(expalin what one and many mean, pros and cons. )
(draw a illustration try to cover the main experimental difference between diff methods)
[1][2] 
### 2.2.1 One-to-one:
- 3C

### 2.2.2 One-to-all:
- Multiplexed 3C-seq
- Open-ended 3C
- 3C-DSL
- 4C
- 4C-seq
- TLA
- e4C
- ACT

### 2.2.3 Many-to-many:
- 5C
- ChIA-PET 

### 2.2.4 Many-to-all
- Capture-3C
- Capture-HiC
- PLAC-seq/HiChIP

### 2.2.5 All to all 
- GCC
- Hi-C
- ELP
- TCC

### Hi-C enrichment
- Single-cell
- In situ/DNase/Micro-HiC

## Another methods for classification:
### Locus-landmark: 
Measure interactions of genomic loci with relatively fixed nuclear ‘landmarks’
- Techniques: ChIP, DamID, TSA-seq

### Locus-locus: 
Measure interactions between genomic loci
- Techniques: 3C, Hi-C, ChIA-PET etc.



**Referrence **
https://doi.org/10.1016/j.csbj.2018.02.003<br>
https://github.com/hms-dbmi/hic-data-analysis-bootcamp/blob/master/HiC-Protocol.pptx 


