# 1.3 From intra to inter chromosomes "talk"

Usually, in order to accomplish various functions for living, a cell nuclei require complex but precise three-dimensional architecture or even four-dimensional variation of chromosome to establish cell-specific gene-expression programs. 

Through out years, lots of endeavors have been made to understand intra-chromosomal interactions by understanding how chromosomes _fold_. This question leads to the discovery of functional and structural elements like TADs, CTCF, cohesin, lncRNAs \(introduced in 1.1\) and so on.

However, it's been a long time since people first raised the hypothesis that chromosomes occupy defined volumes at defined positions, and that they interact with neighboring chromosomes by observing constant number and structural persistence of chromosomes during interphase. \(Carl Rabl 1885, [Cremer et al,. 2006](https://pdfs.semanticscholar.org/b91a/9b9be43ec7e1a10642bf6c44e6a76156637a.pdf)\) . More recently,  advanced technologies both from imaging and high throughput sequencing blaze the trail to understand inter-chromosomes interaction. 

Let's see some real examples while bearing in mind the questions such as: What are the mechanisms of inter chromosomes interactions and what is the driven force of these interactions?  

## Olfactory receptor gene cluster

The sense of smell provides crucial information about the environment of life like order of food or predator. How do our olfactory system decodes the complexity of the real-world odors and discern the differences among tens or hundreds different odorants of a smell?  

It turns out the vast diversity ofodorants detected by the olfactory system is mirrored by a large repertoire of chemosensory proteins. The olfactory system detects odorants with G protein–coupled receptors \(GPCRs\) expressed on the surface of sensory neurons. The olfactory GPCRs belong to several different families, one of them is the olfactory receptors \(ORs\) \([Monahan et al,. 2015](https://www.annualreviews.org/doi/pdf/10.1146/annurev-cellbio-100814-125308)\). 

The OR gene cluster, in which individual non-homologous chromosomal contacts allow the expression of single ORs in each cell to create a diverse repertoire of OR expression at the tissue level. At any given time, only a few of the ∼1,400 OR genes located on 18 different chromosomes converge in the same in ter-chromosomal space \([Maass et al,. 2018](http://jcb.rupress.org/content/218/1/27.abstract)\). 

In situ HiC on FAC-sorted olfactory sensory neurons shows that olfactory receptor \(OR\) genes from numerous chromosomes make frequent, extensive, and highly specific inter-chromosomal contacts that strengthen with differentiation. Moreover, in terminally differentiated olfactory sensory neurons, &gt;30 intergenic enhancers generate a multi-chromosomal hub that associates only with the single active OR from a pool of ~1400 genes \([Horta et al., 2018](https://www.biorxiv.org/content/biorxiv/early/2018/03/23/287532.full.pdf)\). 

![](../.gitbook/assets/image%20%2811%29.png)

**Figure1 Mature Olfactory Sensory Neurons \(mOSNs\) make extensive interchromosomal contacts between olfactory receptor \(OR\) clusters.** 35.6% interchomosomal interactions and ****chromosome 2 and 9 show highly restricted and frequent contacts between OR gene clusters in cis and trans in mOSNs. _**\(figure by Horta, Adan et al,. 2018 Cell type-specific interchromosomal interactions as a mechanism for transcriptional diversity.\)**_



## Transcription factory

Genomic interactions also appear to be influenced by chromosomal location and transcription. The spatial positioning of genes chromosomes in the 3D nucleus can be important for their transcriptional regulation. Nuclear speckles and transcription factories are good examples to be examined and considerable evidence supported \( [Rieder et al,. 2012](https://www.ncbi.nlm.nih.gov/pubmed/23109938) \). 

The term transcription “factory” was first used in [1993 by Jackson and colleagues](https://www.ncbi.nlm.nih.gov/pubmed/8458323). They exploited Br-UTP labeling to visualize mRNA synthesis in permeabilized HeLa cells that were encapsulated inagarose microbeads. Using confocal microscopy to visualize the transcripts, they found that **transcription occurred at 300–500 discrete sites in the nucleus, rather than being homogeneously distributed throughout the nucleus.** Later, [Iborra et al \(1996](http://jcs.biologists.org/content/109/6/1427)\) compared the distribution of transcripts with the distribution of **RNA polymerase II,** found out that ****the clustered polymerase molecules showed a one-to-one co-localizationwith transcripts, suggesting that the clusters contain transcriptionally engaged polymerase. 

The structure of a transcription factory appears to be determined by cell type, transcriptional activity of the cell and also the method of technique used to visualize the structure. The generalised view of a transcription factory would feature between 4 – 30 RNA polymerase molecules and it is thought that the more transcriptionally active a cell is, the more polymerases that will be present in a factory in order to meet the demands of transcription. The core of the factory is porous and protein rich. A factory only contains one type of RNA polymerase and the diameter of the factory varies depending on the RNA polymerase featured; RNA polymerase I factories are roughly 500 nm in width whereas RNA polymerase II and III factories a magnitude smaller at 50 nm. It has been experimentally shown that the transcription factory is immobilised to a structure and it is postulated that this immobilisation is because of a tethering to the [nuclear matrix](https://en.wikipedia.org/wiki/Nuclear_matrix); this is because it has been shown it is tied to a structure that is unaffected by [restriction enzymes](https://en.wikipedia.org/wiki/Restriction_enzymes). \( [Wikipedia transcription factory ](https://en.wikipedia.org/wiki/Transcription_factories)\)

{% embed url="https://www.youtube.com/watch?v=Naa-JwGU6PM" %}

A model of transcription factory from [Peter R Cook's Lab](http://users.path.ox.ac.uk/~pcook/images/pombotext.html), a pioneer raised the idea of transcription factory. 

These transcription factories can take place between expression-dependent loci in cis or between nonhomologous chromosomes where transcriptionally active loci preferentially contact active rather than silent loci \([Hacisuleyman, et al,. 2014](https://www.nature.com/articles/nsmb.2764), [Zhao et al,. 2006](https://www.nature.com/articles/ng1891) \).  

## Reference

\[1\] Maass P G, Barutcu A R, Rinn J L. Interchromosomal interactions: A genomic love story of kissing chromosomes\[J\]. J Cell Biol, 2019, 218\(1\): 27-38.

\*\*\*\*











