# 1.3 From intra to inter chromosomes "talk"

Usually, in order to accomplish various functions for living, a cell nuclei require complex but precise three-dimensional architecture or even four-dimensional variation of chromosome to establish cell-specific gene-expression programs. 

Through out years, lots of endeavors have been made to understand intra-chromosomal interactions by understanding how chromosomes _fold_. This question leads to the discovery of functional and structural elements like TADs, CTCF, cohesin, lncRNAs \(introduced in 1.1\) and so on.

However, it's been a long time since people first raised the hypothesis that chromosomes occupy defined volumes at defined positions, and that they interact with neighboring chromosomes by observing constant number and structural persistence of chromosomes during interphase. \(Carl Rabl 1885, [Cremer et al,. 2006](https://pdfs.semanticscholar.org/b91a/9b9be43ec7e1a10642bf6c44e6a76156637a.pdf)\) . More recently,  advanced technologies both from imaging and high throughput sequencing blaze the trail to understand inter-chromosomes interaction. 

Let's see some real examples while bearing in mind the questions such as: What are the mechanisms of inter chromosomes interactions and what is the driven force of these interactions?  

## Olfactory receptor gene cluster

The sense of smell provides crucial information about the environment of life like order of food or predator. How do our olfactory system decodes the complexity of the real-world odors and discern the differences among tens or hundreds different odorants of a smell?  

It turns out the vast diversity ofodorants detected by the olfactory system is mirrored by a large repertoire of chemosensory proteins. The olfactory system detects odorants with G protein–coupled receptors \(GPCRs\) expressed on the surface of sensory neurons. The olfactory GPCRs belong to several different families, one of them is the olfactory receptors \(ORs\) \([Monahan et al,. 2015](https://www.annualreviews.org/doi/pdf/10.1146/annurev-cellbio-100814-125308)\). 

The OR gene cluster, in which individual non-homologous chromosomal contacts allow the expression of single ORs in each cell to create a diverse repertoire of OR expression at the tissue level. At any given time, only a few of the ∼1,400 OR genes located on 18 different chromosomes converge in the same in ter-chromosomal space \([Maass et al,. 2018](http://jcb.rupress.org/content/218/1/27.abstract)\). 

In situ HiC on FAC-sorted olfactory sensory neurons shows that olfactory receptor \(OR\) genes from numerous chromosomes make frequent, extensive, and highly specific inter-chromosomal contacts that strengthen with differentiation. Moreover, in terminally differentiated olfactory sensory neurons, &gt;30 intergenic enhancers generate a multi-chromosomal hub that associates only with the single active OR from a pool of ~1400 genes \([Horta et al., 2018](https://www.biorxiv.org/content/biorxiv/early/2018/03/23/287532.full.pdf)\). 

![](../.gitbook/assets/image%20%2811%29.png)

**Figure1 Mature Olfactory Sensory Neurons \(mOSNs\) make extensive interchromosomal contacts between olfactory receptor \(OR\) clusters.** 35.6% interchomosomal interactions and ****chromosome 2 and 9 show highly restricted and frequent contacts between OR gene clusters in cis and trans in mOSNs. _**\(figure by Horta, Adan et al,. 2018 Cell type-specific interchromosomal interactions as a mechanism for transcriptional diversity.\)**_











## Reference







