# 4.1 Experimental Methods

## One-to-all capturing

These RNA specific mapping technologies are based on utilizing biotinylated complementary oligonucleotides to pull down a specific target RNA together with its binding partners. The identities of its DNA or protein-binding partners are subsequently revealed by sequencing or mass spectrometry. Typical technologies includes:

* Chromatin Isolation by RNA Purification [\(ChIRP\)](https://www.ncbi.nlm.nih.gov/pubmed/21963238)
* Capture Hybridization Analysis of RNA Targets [\(CHART\)](https://www.ncbi.nlm.nih.gov/pubmed/22143764)
* RNA Antisense Purification [\(RAP\)](https://www.ncbi.nlm.nih.gov/pubmed/23828888)

All these three methods leveraging synthetic biotinylated antisense DNA oligonucleotides that can specifically bind to genome, and later lncRNA-chromatin complex will be pulled out and sequenced subsequently. These three methods are very similar in overall approach with differences in specific crosslinking reagents, strength of chromatin shearing, strength of washing buffer, density and length of antisense probes. A review can be found \( RNA, action through interactions. _Trends in genetics_\)

## Global mapping

Methods mentioned above allow analysis of only one known RNA at a time. A new cohort of methods take advantage of the proximity of RNA and it's targets. By using a bi-polar linker which has RNA linker at one end and DNA linker at the other, these methods conversing the proximity into chimeric sequence which can later be mapped to referrence and find the contacts. These comprehensive and unbiased repertoires maybe useful in de novo interaction detection.

[**MARGI**](http://dx.doi.org/10.1016/j.cub.2017.01.011)**: Mapping RNA-genome interactions** 

> MARGI \(Sridhar et al., 2017\) is the first proposed method for capturing global relationship of RNA-genome. The major innovation of the MARGI technology is to ligate chromotin associated RNAs with their target genomic sequences by proximity ligation, forming RNA-DNA chimeric sequences, which are subsequently converted into a sequencing library and subjected to paired end sequencing. MARGI has two variations: one is pxMARGI \(proximity MARGI\), this version is designed to capture _all_ spacial approximate RNA-genome interaction no matter there is a protein binding or not. However, another one is diMARGI \(direct MARGI\) which only aim to detect protein-mediated RNA-DNA connection. The prominent experimental difference is weather we use sonication

[**CHAR-seq**](https://elifesciences.org/articles/27024
http://www.ncbi.nlm.nih.gov/pubmed/29648534)**: Chromatin-associated RNA sequencing**

> CHAR-seq \(Bell et al.2018\) maps all RNA-to-DNA contacts across the genome. Briefly, cells are cross-linkedfirst, then RNA is partially fragmented and soluble RNA is washed away. The chromatin-cross-linked RNA is then ligated to an oligonucleotide duplex ’bridge’ molecule and reverse transcribed. Genomic DNA is then digested and ligated onto the other end of the oligonucleotide ’bridge’, creating a link between chromatin-associated RNA and proximal DNA. The ligated RNA is fully converted to cDNA during second strand synthesis. Finally, the DNA is sonicated and the chimeric molecules are purified, processed, and sequenced.

 ****[**GRID-seq**](http://dx.doi.org/10.1038/nbt.3968)**: Global RNA interactions with DNA by deep sequencing**:

> GRID-seq \(Li et al. 2017\) is also a global mapping method that utilize a bivalent linker to ligate RNA to DNA in situ on fixed nuclei. One key difference with other two technologies is that GRID-seq uses a restriction enzyme to cut cDNA and gDNA fragments 19–23 bp distal to the bridge, which allows size selection of molecules that contain both RNA and DNA. This likely reduces the number of uninformative molecules sequenced.

### Differences among three methods:

**1\) linker differences**

As we described before, the interaction between proximate RNA-DNA interaction is transformed by using a linker to connect RNA at one end and DNA at the other. However the design of the **linker** \(also names as _bridge_\) varies between three methods the basic scheme of a linker follow a same fashion. Here we illustrate MARGI's linker construction as an example:

![Figure1 MARGI linker illustration](../.gitbook/assets/margi_adaptor%20%281%29.png)

Usually each sequence of the double strand linker varies,  one has a 5' adenylated end that aims to ligate RNA and the other one has a notch for DNA ligation. The chimeric sequence will then be circulated and split by enzyme. The right and left flanking sequences of the linker are designed to be complementary to the SP1 and SP2 sequencing primers, which enables the product sequences originated from the RNA and the DNA to be sequenced as Read 1 and Read 2, respectively, from the two distinct sequencing primers by paired-end sequencing.

|  | MARGI | CHAR-SEQ | GRID-SEQ |
| :--- | :---: | :---: | :---: |
| Sequencing strategy | PE | SE | PE |
| Enzyme | BamHI | Dpnll | Mmel |
| Reads length | customize\* | 152bp | customize\* |

* PE: pair-end sequencing.   SE: single-end sequencing.  
* Customize means the length of reads depends on different sequencing platforms. 

**2\) Protocol differences**

The major difference is that many steps are conducted in intact nuclei, including restriction enzyme digestion, RNA-linker ligation, and proximity ligation. 

![Figure2 Protocal overview of three global sequencing-based technologies.](../.gitbook/assets/figure3_withtable_v2.png)

* Proximity ligations of CHAR-seq and GRID-seq are performed in situ in intact nuclei, which reduces nonspecific interactions [\(Rao et al., 2014\)](https://www.ncbi.nlm.nih.gov/pubmed/25497547).
* ChAR-seq uses long single-end reads to sequence across the entire junction of the ‘bridge’, ensuring that RNA-to-DNA contacts are mapped with high confidence but the long reads requirement may reduce some of the informative information. 

<table>
  <thead>
    <tr>
      <th style="text-align:left"></th>
      <th style="text-align:center">Advantages</th>
      <th style="text-align:center">Disadvantages</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align:left">MARGI</td>
      <td style="text-align:center">
        <p></p>
        <p>&#x25CF; Captures interaction at native conditions.</p>
      </td>
      <td style="text-align:center">&#x25CF; Require a large number (10^7) of cells.</td>
    </tr>
    <tr>
      <td style="text-align:left">CHAR-seq</td>
      <td style="text-align:center">
        <p></p>
        <p>&#x25CF; Proximity ligation is performed in nuclei, which reduces nonspecific
          interactions.(in situ)</p>
      </td>
      <td style="text-align:center">&#x25CF; Only sequencing reads that cover the entire bridge sequence are
        informative, reducing the number of informative reads.</td>
    </tr>
    <tr>
      <td style="text-align:left">GRID-seq</td>
      <td style="text-align:center">&#x25CF; in situ, reduce nonspecific interactions.</td>
      <td style="text-align:center">&#x25CF; The informative sequence lengths on the RNA side and the DNA
        side are both limited to ~20 bases, resulting in challenges in unambiguous
        sequence mapping.</td>
    </tr>
  </tbody>
</table>













