# Chap4 RNA-genome interaction

## Introduction

RNA plays a critical component of chromatin in eukaryotes not only as transcription product, but also a decoration on chromatin as a new layer of epigenome. RNAs regulate both local and global chromatin states that regulate the genome activity of life. To investigate the interaction between RNAs and genome, methods have been evolving from one-to-all \(capturing all targets of one specific RNA\) to global methods that investigate all-RNA-versus-genome interactions. Here we'll give an overview of different methods and try to elucidate the differences. Also, some of the leading-edge discoveries will be discussed to possibly give you more insights for in-depth understanding of 3D genome interaction and regulation.

## Some discoveries

### 1. Two kinds of RNA-DNA interaction

Even though the nucleus is the largest cellular [organelle](https://en.wikipedia.org/wiki/Organelle) in animal cells \( Molecular Cell Biology \(5th ed.\),  the average diameter of the nucleus is only approximately 6 \(µm\). All the materials are highly compacted and condensed in such a limited space. As people could imagine, magic would happen there. In our scenario, RNA and DNA interaction can be caused not only by a protein-mediated **direct interaction**, but also physical spacial approximate - **proximal interaction.**  The direct interaction is usually robust to sonication, and people utilize this feature to distinguish from the second one. Proximal interaction is sometimes caused just by diffusion movement, others may like nascent RNAs. By leveraging MARGI, researchers have found:

* The largest components of non-coding pxRNAs were [pseudogene](https://en.wikipedia.org/wiki/Pseudogene), antisense, and long intergenic noncoding RNA \([lincRNA](https://www.biosyn.com/tew/what-are-lncrnas-and-lincrnas.aspx)\).
* The largest component of diRNAs in both cell types was small nucleolar RNA \([snoRNA](https://en.wikipedia.org/wiki/Small_nucleolar_RNA)\).



### 2. Distribution of RNA-DNA interaction over distance

If we take a closer look at all the interaction pairs, examine the distance between RNA and DNA locations, there is an inverse correlation between distance and contacts were observed by different paper [\[1\]](https://linkinghub.elsevier.com/retrieve/pii/S0960982217300118)[\[2\]](http://dx.doi.org/10.1038/nbt.3968). 

Even though the criteria of different paper may vary, similar results were found between different groups. Sridhar et al. found that most where approximately 80% were proximal \(one the same chromosome within 2k bp\) , less than 4% were distal \(one the same chromosome more than 2000bp\), and 15%–20% were inter-chromosomal, suggesting a non-trivial amount of long-range interactions. 



### 3. Putative enhancers 

Enhancer RNAs \(Sigova et al., 2015\), have been shown to regulate expression of specific genes by localizing to chromatin and recruiting activating or repressing proteins. Recent studies also suggest that enhancers may be segregated into typical and super-enhancers, the latter being defined by a much higher density of enhancer marks. People found that enhancers are highly associated with RNAs predominantly corresponded to super-enhancers in some cells.





































































































































































